src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,250 @@
+package mate.jdbc.dao.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import mate.jdbc.dao.CarDao;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String insertRequest = "INSERT INTO cars (manufacturer_id, model) "
+                + "VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(insertRequest,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, car.getManufacturer().getId());
+            statement.setString(2, car.getModel());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create a car: "
+                    + car + ". ", e);
+        }
+        setCarToDriversRelation(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getRequest = "SELECT c.id AS car_id, c.model, m.id AS manufacturer_id, "
+                + "m.name AS manufacturer_name, m.country, "
+                + "d.id AS driver_id, d.name AS driver_name, d.license_number "
+                + "FROM cars c "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "JOIN cars_drivers cd ON c.id = cd.cars_id "
+                + "JOIN drivers d ON d.id = cd.drivers_id "
+                + "WHERE c.id = ? AND c.is_deleted = FALSE AND d.is_deleted = FALSE;";
+        Car foundedCar = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(getRequest)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                foundedCar = (foundedCar == null)
+                        ? buildCarFromResultSet(resultSet)
+                        : addDriverToCar(buildDriverFromResultSet(resultSet), foundedCar);
```

Comment:
nested connections problem, refer to common mistakes. To fix it let's add drivers out of this try/catch block